apiary.apib

```diff
@@ -3281,6 +3281,312 @@ Update the current timer. Only possible if there is a timer running.
+            + department (object)
+                + type: `department` (string)
+                + id: `5e90eb0a-b502-4344-aa0f-3b8525af6186` (string)
+            + layout (object)
+                + type: `layout` (string)
+                + id: `a05320f6-b3ef-4ff9-a287-2074a12fb3e9` (string)
+            + milestone (object)
+                + type: `milestone` (string)
+                + id: `e94c1363-3426-46a4-921d-fce03f39bbd5` (string)
+            + event (object)
+                + type: `event` (string)
+                + id: `9fc14045-5c6c-4ba8-8672-42ea3f26aa63` (string)
+            + location (Address)
+            + distance (Distance)
+            + created_at: `2019-05-09T11:25:11+00:00` (string)
+            + updated_at: `2019-05-09T11:30:58+00:00` (string)
+            + time (array)
+                + (object)
+                    + user (object)
+                        + type: `user` (string)
+                        + id: `4ec55a8c-2d80-472a-a9c2-5ff5ee7eac6a` (string)
+                    + started_at: `2017-04-26T10:01:49+00:00` (string)
+                    + duration: 3600 (number) - In seconds
+                    + description (string)
+            + materials (array)
+                + (object)
+                    + product (object)
+                        + type: `product` (string)
+                        + id: `e2314517-3cab-4aa9-8471-450e73449040` (string)
+                    + quantity: 3 (number)
+            + images (array) - Only included with request parameter `includes=images`.
+                + (object)
+                    + id: `a856642a-a476-4b69-8e42-e7d1badb2ce9` (string)
+                    + location: `https://cdn.teamleader.eu/file` (string) - A temporary URL where the image can be downloaded
+                    + expires: `2018-02-05T16:44:33+00:00` (string) - Expiration time of the temporary download URL
+            + custom_fields (array[CustomField])
+
+### workOrders.attachImage [POST /workOrders.attachImage]
+
+Attach image to a draft work order.
+
++ Request (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + id: `4afb0a9c-91c6-49ed-a2e5-ce7c1e3a87fb` (string, required)
+        + image: `data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAIAAACQkWg2AAAAAXNSR0IArs4c6QAAAVlpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8AAAASUVORK5CYII=` (string, required) - PNG image, base64 encoded
+
++ Response 200 (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + data (object)
+            + type: `workOrderImage` (string)
+            + id: `a856642a-a476-4b69-8e42-e7d1badb2ce9` (string)
+
+### workOrders.deleteImage [POST /workOrders.deleteImage]
+
+Delete image from a draft work order.
+
++ Request (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + id: `4afb0a9c-91c6-49ed-a2e5-ce7c1e3a87fb` (string, required)
+        + image_id: `a856642a-a476-4b69-8e42-e7d1badb2ce9` (string, required)
```

Comment:
Q: isn't one id enough? (the id of the work order image?)

Reply:
If we keep both it's clear that you can't pass any file ID there and expect it to work. You need an actual WorkOrder ID.

Although it should be clear from the endpoint name as well. So not sure what to decide here.

Reply:
Both are okay for me, just seemed like obsolete data, but you're probably never going to have the file id without the work order id anyway